### DISCUSSION TITLE
Long-running background queries for object metadata make UI drag when exploring Vertica.

### DISCUSSION THREAD

**Discussion by bitadmiral** [dbeaver/dbeaver#14611] [Q&A]
Hi, I recently upgraded from DBeaver 4 to 21.2.4.202111011413, and find the UI to be very slow and even unresponsive compared to the older version.  I believe it has to do with the fact that the modern DBeaver gathers much more metadata on the database objects (which is nice).  I've traced the problem to background queries that retrieve object metadata, specifically any query that involves Vertica's **comments** system view.   In our databases, any query on the **comments** view takes a number of minutes to complete.  Such queries are executed in the background by DBeaver often, including for autocompletion or exploring objects in the Database Navigator.

We actually don't use object comments at all.  But even a query that attempts to retrieve a single row from the **comments** view takes many minutes.  It is because that system view is extremely complex, with many joins, unions, subqueries on other views, etc.  This causes strange behaviors in DBeaver, such as popup dialogs popping up at seemingly random times, long waits for auto-completion, and temporary freezes in the UI.

The problem is detailed in my post in the Vertica forum here:  [Querying v_catalog.comments is very slow even with no results](https://forum.vertica.com/discussion/242480/querying-v-catalog-comments-is-very-slow-even-with-no-results)

I'm wondering if there is a way to disable retrieval of object comments from the Vertica database completely.  Or, maybe it would be possible to add an option to disable retrieval of object comments, and eliminate all references to the **comments** view in all background queries.

BTW, this is somewhat related to an old issue:  #3474

**Comment by LonwoLonwo** [comment]
Hello @verticauser810 

DBeaver doesn't have such of setting to avoid comments reading.
So please convert your discussion into a feature request and we will add this ticket to our working plans.

**Reply by bitadmiral** [reply]
Okay, thank you Nastia.  I have created #14625 .  Please let me know if I should alter the request in any way to improve the workflow.
